### PR TITLE
feature:add more test case for codeGenerateController and pipelineClient

### PR DESCRIPTION
**Pull request by zzvswxy**
feature:
  add more test case for codeGenerateController and pipelineClient
  add mockserver for UT

Signed-off-by: zhouzheng <576014838@qq.com>

### CODE REVIEW

File: src/codegenApp.ts [hunk 1]
```diff
@@ -41,6 +43,10 @@ class CodegenApp {
         this.buildSchedulerTask();
     }
 
+    public async shutdown(): Promise<void> {
+        await this.mongoDbConnection.close();
```
Comment (chunyu3) [inline src/codegenApp.ts:47]: can mongoDbConnection be undefined?

File: test/controller/codeGenerateController.test.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        await server.start();
+    });
+    afterAll(async () => {
+        await server.stop();
+        await codegenAppClient.shutdown();
+    });
+    beforeEach(() => server.reset());
+
+    test('codeGenerateController test basic api', async () => {
+        const mock = server.post('/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_apis/pipelines/14243/runs').mockImplementation((ctx) => {
```
Comment (chunyu3) [inline test/controller/codeGenerateController.test.ts:40]: Only test the Rest API call ( the reset api exists and works), but does not test the business logic of each API.

File: test/controller/codeGenerateController.test.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        response = await axios.post('http://localhost:3000/codegenerations/controllerTest/cancel');
+        expect(response.status).toBe(200);
+
+        response = await axios.post('http://localhost:3000/codegenerations/controllerTest/complete');
+        expect(response.status).toBe(200);
+
+        response = await axios.get('http://localhost:3000/codegenerations/');
+        expect(response.data.length).toBeGreaterThan(0);
+
+        response = await axios.delete('http://localhost:3000/codegenerations/controllerTest');
```
Comment (chunyu3) [inline test/controller/codeGenerateController.test.ts:142]: Need to check if the codegeneration is really deleted. not just response code

File: test/controller/pipelineClient.test.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            specRepoUrl: 'string',
+            specRepoBaseBranch: 'string',
+            sdkRepoType: 'string',
+            sdkRepoUrl: 'string',
+            sdkRepoBaseBranch: 'string',
+            skippedTask: 'string',
+            serviceType: 'string',
+        });
+
+        expect(response.status).toEqual(200);
```
Comment (chunyu3) [inline test/controller/pipelineClient.test.ts:89]: Need to check if the pipeline build is created

File: src/codegenApp.ts (hunk 1 above)
Comment (zzvswxy) [reply src/codegenApp.ts:47]: > can mongoDbConnection be undefined?

if serves can't connect mongodb, 'buildMongoDBCollection()' in codegenApp.ts will throw an error

File: test/controller/codeGenerateController.test.ts (hunk 3 above)
Comment (zzvswxy) [reply test/controller/codeGenerateController.test.ts:182]: > Need to check if the codegeneration is really deleted. not just response code

fixed, thx

File: test/controller/pipelineClient.test.ts (hunk 4 above)
Comment (zzvswxy) [reply test/controller/pipelineClient.test.ts:89]: > Need to check if the pipeline build is created

pipeline server is mocked by jest-mock-server
use 'expect(mock).toHaveBeenCalledWith' to check the api is called

**Review by chunyu3 — APPROVED** [review]

**Pull request merged**